indico/modules/categories/util.py

```diff
@@ -229,3 +230,8 @@ def serialize_category_role(role, legacy=True):
             'name': role.name,
             'identifier': f'CategoryRole:{role.id}',
         }
+
+
+def can_create_unlisted_events(user):
+    return (user.is_admin or unlisted_events_settings.get('enabled') and (not unlisted_events_settings.get('restricted')
+            or unlisted_events_settings.acls.contains_user('authorized_creators', user)))
```

Comment:
I think this will be much more readable when splitting it into multiple statements:

```python
def can_create_unlisted_events(user):
    if not user or not unlisted_events_settings.get('enabled'):
        return False
    elif user.is_admin or not unlisted_events_settings.get('restricted'):
        return True
    else:
        return unlisted_events_settings.acls.contains_user('authorized_creators', user)
```

Reply:
and maybe a small unit test for that function would be nice